src/view/utils/MessageUtils.tsx

```diff
@@ -0,0 +1,9 @@
+interface vscode {
+    postMessage(message: any): void;
+}
+
+declare const vscode: vscode;
+
+export const sendMessage = (type: string, state: any) => {

```

Comment:
I'd use generics here:

export const sendMessage\<TState\> = (type: string, state: TState) => { ... }

Reply:
Good tip, changed to generics